test/NuGet.Clients.Tests/NuGet.SolutionRestoreManager.Test/VsSolutionRestoreServiceTests.cs

```diff
@@ -2694,6 +2694,51 @@ public void ToPackageSpec_WithCPMAndVersionOverride_SetsVersionToVersionOverride
             tfm.Dependencies[1].AutoReferenced.Should().BeFalse();
         }
 
+        [Fact]
+        public void ToPackageSpec_WithCPMAndPackageReferenceWithDifferentCasing_CombinesDataCorrect()
+        {
+            // Arrange
+            ProjectNames projectName = new ProjectNames(@"f:\project\project.csproj", "project", "project.csproj", "projectC", Guid.NewGuid().ToString());
+            var emptyReferenceItems = Array.Empty<VsReferenceItem>();
+            var projectProperties = ProjectRestoreInfoBuilder.GetTargetFrameworkProperties(CommonFrameworks.NetStandard20);
+            projectProperties[ProjectBuildProperties.ManagePackageVersionsCentrally] = "true";
+            projectProperties[ProjectBuildProperties.CentralPackageVersionOverrideEnabled] = "true";
+            string packageId = "foo";
+
+            var targetFrameworks = new VsTargetFrameworkInfo4[] { new VsTargetFrameworkInfo4(
+                items: new Dictionary<string, IReadOnlyList<IVsReferenceItem2>>()
+                {
+                    [ProjectItems.PackageReference] = [new VsReferenceItem2(packageId, EmptyProperties)],
+                    [ProjectItems.PackageVersion] =
+                    [
+                        new VsReferenceItem2(packageId.ToUpperInvariant(), new Dictionary<string,string>() { ["Version"] = "2.0.0"}),
+                    ]
+                },
+                properties: projectProperties)
+            };
+
+            var pri = new VsProjectRestoreInfo3
+            {
+                MSBuildProjectExtensionsPath = string.Empty,
+                TargetFrameworks = targetFrameworks,
+                OriginalTargetFrameworks = CommonFrameworks.NetStandard20.ToString(),
+            };
+
+            // Act
+            var result = VsSolutionRestoreService.ToPackageSpec(projectName, pri);
+
+            // Assert
+            var tfm = result.TargetFrameworks.First();
+
+            tfm.CentralPackageVersions.Should().HaveCount(1);
+            result.RestoreMetadata.CentralPackageVersionsEnabled.Should().BeTrue();
+            tfm.Dependencies.Should().HaveCount(1);
+            tfm.Dependencies[0].Name.Should().Be(packageId);
```

Comment:
I thought this was checking the case of packageId, but I tested this and actually this was already correct before this PR ("foo" in both before and after this PR). The key is the following assertion of `VersionCentrallyManaged` being true. Maybe the test name should reflect that.
eg,
`ToPackageSpec_WithCPMAndPackageReferenceWithDifferentCasing_VersionCentrallyManaged()`

Reply:
It's the same thing as the others, the data wasn't combined correctly, I think that's a better name. 
It's VersionCentrallyManaged & the actual Version.
Everything else should be true cased.